addons/point_of_sale/models/account_journal.py

```diff
@@ -1,10 +1,16 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 # Copyright (C) 2004-2008 PC Solutions (<http://pcsol.be>). All Rights Reserved
-from odoo import fields, models, api
-
+from odoo import fields, models, api, _
+from odoo.exceptions import ValidationError
 
 class AccountJournal(models.Model):
     _inherit = 'account.journal'
 
     pos_payment_method_ids = fields.One2many('pos.payment.method', 'journal_id', string='Point of Sale Payment Methods')
+
+    @api.constrains('type')
+    def _check_type(self):
+        methods = self.env['pos.payment.method'].sudo().search([("journal_id", "=", self.id)])
```

Comment:
does self.id is always set? (when you create a new record)

Because if not, you'll look for payment method having no journal_id which can be the case when you have set it up to "pay later".

It'will not let you create any account journal if you have a payment method without journal_id

Reply:
Is it even possible to have a record that doesn't have an Id automatically assigned? Because I don't know in what case a journal wouldn't have an Id

Reply:
@robinengels When you create a new one. If the constraint is triggerd before the "creation" (which I am not sure), there could be no ID

Reply:
@robinengels I've tested it, and it seems that id is already set when the constraint is triggered. So it is ok

Reply:
good to know